src/coniql/app.py

```diff
@@ -1,49 +1,82 @@
-import traceback
+import logging
 from argparse import ArgumentParser
 from pathlib import Path
-from typing import Any, Dict
+from typing import List, Optional
 
 import aiohttp_cors
+import strawberry
 from aiohttp import web
-from tartiflette import Engine, TartifletteError
-from tartiflette_aiohttp import register_graphql_handlers
+from graphql import GraphQLError
+from strawberry.aiohttp.views import GraphQLView
+from strawberry.subscriptions import GRAPHQL_TRANSPORT_WS_PROTOCOL, GRAPHQL_WS_PROTOCOL
+from strawberry.types import ExecutionContext
 
-from coniql.caplugin import CAPlugin
-from coniql.plugin import PluginStore
-from coniql.pvaplugin import PVAPlugin
-from coniql.simplugin import SimPlugin
+import coniql.strawberry_schema as schema
 
 from . import __version__
 
 
-async def error_coercer(exception: Exception, error: Dict[str, Any]) -> Dict[str, Any]:
-    if isinstance(exception, TartifletteError):
-        e = exception.original_error
+class ConiqlSchema(strawberry.Schema):
+    # Override to remove stack trace printer with every log.
+    # Note that is mechanisms is undocumented and could break if Strawberry
+    # update the underlying code this is based on.
+    def process_errors(
+        self,
+        errors: List[GraphQLError],
+        execution_context: Optional[ExecutionContext] = None,
+    ) -> None:
+        for error in errors:
+            if not error.message:
+                logging.error("Unknown error occurred. Enable debugging to find cause.")
```

Comment:
I would like to suggest a different mechanism for this.

The problem with using the default logging functions `logging.error` is that you are using the root logger rather than a specific one, and you therefore lose the ability to control these log messages with app-level config later. In the base implementation, they use the `StrawberryLogger` class which in turn uses a named logger `strawberry.execution`.

In the base implementation `StrawberryLogger.error` accepts a keyword argument `stack_info` which can be set to False. So one option would be to override this with that argument set to False:

```
    def process_errors(
        self,
        errors: List[GraphQLError],
        execution_context: Optional[ExecutionContext] = None,
    ) -> None:
        for error in errors:
            StrawberryLogger.error(error, execution_context, stack_info=False)
```

Better yet would be to configure logging at the app level in `coniql.app.main`, and define a custom `logging.Formatter` which overrides the https://docs.python.org/3/library/logging.html#logging.Formatter.formatStack formatStack method to suppress the exception information (perhaps conditional on the coniql Debug argument). This would be more resilient to changes of undocumented Strawberry behaviour.

Reply:
Thanks for the suggestion and demo code. I have implemented the latter option using a custom logging formatter to suppress the exception stacktrace in commit: https://github.com/dls-controls/coniql/pull/44/commits/61fe51d70f0370cc37ff6819ca21f04d0bab5102. 
This eliminates the need to overwrite the undocumented Strawberry method.